contrib/Shopify/sarama/option.go

```diff
@@ -22,16 +27,23 @@ func WithServiceName(name string) Option {
 
 // WithAnalytics enables Trace Analytics for all started spans.
 func WithAnalytics(on bool) Option {
-	if on {
-		return WithAnalyticsRate(1.0)
+	return func(cfg *config) {
+		if on {
+			cfg.analyticsRate = 1.0
```

Comment:
It's confusing to me that you have not used the same style anymore:
```go
if on {
    return WithAnalyticsRate(1.0)
else {
    return WithAnalyticsRate(math.NaN())
}
```
Any reason for it?

Reply:
No great reason. It's a one-liner either way, but saves a first-time reader having to jump from here to another function to see what `WithAnalyticsRate` does, which is essentially that assignment.

Reply:
I agree with the "first time reader" argument. I disagree however because I'd prefer to have only one way to set the configuration flag. But I'm fine if you leave it like this. I'll approve the PR and let you decide. Ok?